packages/messaging/ios/src/RCTAEPMessaging.m

```diff
@@ -24,5 +39,127 @@ - (dispatch_queue_t)methodQueue
 RCT_EXPORT_METHOD(extensionVersion: (RCTPromiseResolveBlock) resolve rejecter:(RCTPromiseRejectBlock) reject) {
     resolve([AEPMobileMessaging extensionVersion]);
 }
+
+RCT_EXPORT_METHOD(refreshInAppMessages) {
+    [AEPMobileMessaging refreshInAppMessages];
+}
+
+RCT_EXPORT_METHOD(setMessagingDelegate) {
+    [AEPMobileCore setMessagingDelegate: self];
+}
+
+RCT_EXPORT_METHOD(show: (NSString *) messageId) {
+    AEPMessage * message = [cachedMessages objectForKey:messageId];
+    [message show];
+}
+
+RCT_EXPORT_METHOD(dismiss: (NSString *) messageId) {
+    AEPMessage * message = [cachedMessages objectForKey:messageId];
+    [message dismiss];
+}
+
+RCT_EXPORT_METHOD(track: (NSString *) messageId withInteraction: (NSString *) interaction eventType: (int) eventTypeValue) {
+    AEPMessage * message = [cachedMessages objectForKey:messageId];
+    AEPMessagingEdgeEventType messagingEdgeEventType = -1;
+    
+    if(eventTypeValue == 0){
+        messagingEdgeEventType = AEPMessagingEdgeEventTypeInappDismiss;
+    } else if (eventTypeValue == 1) {
+        messagingEdgeEventType = AEPMessagingEdgeEventTypeInappInteract;
+    } else if (eventTypeValue == 2) {
+        messagingEdgeEventType = AEPMessagingEdgeEventTypeInappTrigger;
+    } else if (eventTypeValue == 3) {
+        messagingEdgeEventType = AEPMessagingEdgeEventTypeInappDisplay;
+    } else if (eventTypeValue == 4) {
+        messagingEdgeEventType = AEPMessagingEdgeEventTypePushApplicationOpened;
+    } else if (eventTypeValue == 5) {
+        messagingEdgeEventType = AEPMessagingEdgeEventTypePushCustomAction;
+    }
+    
+    if(messagingEdgeEventType != -1){
+        [message trackInteraction:interaction withEdgeEventType:messagingEdgeEventType];
+    }
+}
+
+RCT_EXPORT_METHOD(handleJavascriptMessage: (NSString *) messageId messageName: (NSString *) name resolver: (RCTPromiseResolveBlock) resolve rejector:(RCTPromiseRejectBlock) reject) {
+    
+    AEPMessage * message = [cachedMessages objectForKey: messageId];
+    [message handleJavascriptMessage:name withHandler:^(id result) {
+        if(result) {
+            resolve(result);
+        } else {
+            reject(@"error", @"error in handleJavaScriptMessage", nil);
+        }
+    }];
+}
+
+RCT_EXPORT_METHOD(clearMessage: (NSString *) messageId) {
+    [cachedMessages removeObjectForKey:messageId];
+}
+
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(shouldShowMessage: (BOOL) shouldShowMessage) {
+    self->shouldShowMessage = shouldShowMessage;
```

Comment:
any particular reason why you're using `->` notation here instead of `.`?
e.g. - `self.shouldShowMessage = shouldShowMessage;`